web-client/src/components/DropZone.jsx

```diff
@@ -0,0 +1,239 @@
+    },
+    dropMessage: {
+      textAlign: "center",
+      color: "#aaaaaa",
+      fontFamily: "Arial",
+      fontSize: "20px",
+    },
+    uploadIcon: {
+      width: "50px",
+      height: "50px",
+      background: `url(${uploadIcon}) no-repeat center center`,
+      backgroundSize: "100%",
+      textAlign: "center",
+      margin: "0 auto",
+      paddingTop: "30px",
+    },
+    message: {
+      color: "red",
+      padding: "2%",
+      fontSize: "15px",
+    },
+    inputInfo: {
+      height: "10%",
+      width: "100%",
+      textAlign: "left",
+      paddingLeft: "5%",
+      color: "#aaaaaa",
+    },
+    btn: {
+      textTransform: "none",
+    },
+    logo: {
+      height: "30px",
+      margin: "2%",
+      float: "left",
+    },
+    fileName: {
+      height: "30px",
+      width: "60%",
+      margin: "3%",
+      float: "left",
+      textOverflow: "ellipsis",
+      whiteSpace: "nowrap",
+      overflow: "hidden",
+    },
+    fileSize: {
+      height: "30px",
+      margin: "3% 2%",
+      float: "left",
+    },
+    exit: {
+      height: "30px",
+      margin: "3% 2%",
+      float: "right",
+      cursor: "pointer",
+    },
+    dialog: {
+      display: "none",
+    },
+  });
+
+  const style = styles();
+
+  const fileInputRef = useRef();
+
+  const maxFileSize = 20971520; // 1024 * 1024 * 20 = 20 MB
+
+  let file = null;
+  const [message, SetMessage] = useState(null);
+  const [fileName, SetFileName] = useState(null);
+  const [fileSize, SetFileSize] = useState(null);
+  const [divStyle, SetDivStyle] = useState({
+    display: "none",
+    textAlign: "left",
+  });
+
+  const DragOver = (e) => {
+    e.preventDefault();
+  };
+
+  const DragEnter = (e) => {
+    e.preventDefault();
+  };
+
+  const DragLeave = (e) => {
+    e.preventDefault();
+  };
+
+  const FileDrop = (e) => {
+    e.preventDefault();
+    const f = e.dataTransfer.files[0];
+    AddFile(f);
+  };
+
+  const AddFile = (f) => {
+    const isFileValid = ValidateFile(f);
+    const isSizeValid = ValidateSize(f);
+
+    if (!isSizeValid && !isFileValid) {
+      ShowMessage(
+        "Too large file and invalid format. Only PDF file not greather than 20 MB can be uploaded."
+      );
+    } else if (!isFileValid) {
+      ShowMessage("Invalid file format. Only PDF file format can be uploaded.");
+    } else if (!isSizeValid) {
+      ShowMessage(
+        "Too large file. Only file not greather than 20 MB can be uploaded."
+      );
+    } else {
+      file = f;
+      SetFileName(file.name);
+      SetFileSize(FileSize(file.size));
+      SetDivStyle({ display: "block", textAlign: "left" });
+      props.SetFile(file);
+    }
+  };
+
+  const ShowMessage = (message) => {
+    SetMessage(message);
+    setTimeout(() => {
+      SetMessage(null);
+    }, 4000);
+  };
+
+  const ValidateFile = (f) => {
+    const validTypes = ["application/pdf"];
+    if (validTypes.indexOf(f.type) === -1) {
+      return false;
+    }
+    return true;
+  };
```

Comment:
It checks, if the file's type is a PDF.
indexOf returns index (In this case === 1) or -1 (if type of selected file is not a PDF). But I think, it can be done easier.

Reply:
Ouh... I think you mean: `const ValidateFile = (f) => "application/pdf" === f.type;`

Reply:
And yeah, it's easier than my proposition :p